22:09:09.008 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.022 [Builder-OrderedExecutor-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
java.lang.NullPointerException: null
	at org.apache.bookkeeper.client.PendingAddOp.writeComplete(PendingAddOp.java:290) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$null$15(MockBookKeeperTestCase.java:622) ~[test-classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.032 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.078 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.586 [Builder-OrderedExecutor-1-0] ERROR org.apache.bookkeeper.client.LedgerHandle - Closing ledger 1 due to WriteException: Write failed on bookie
22:09:09.587 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.client.PendingAddOp - Write of ledger entry to quorum failed: L1 E99
22:09:09.600 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.603 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.605 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.608 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.611 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.614 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.617 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.622 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.626 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.630 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.639 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.642 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.646 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
22:09:09.707 [bk-test-OrderedScheduler-1-0] ERROR org.apache.bookkeeper.common.util.SafeRunnable - Unexpected throwable caught 
io.netty.util.IllegalReferenceCountException: refCnt: 0, increment: 1
	at io.netty.util.internal.ReferenceCountUpdater.retain0(ReferenceCountUpdater.java:123) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.internal.ReferenceCountUpdater.retain(ReferenceCountUpdater.java:110) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at io.netty.util.AbstractReferenceCounted.retain(AbstractReferenceCounted.java:61) ~[netty-common-4.1.68.Final.jar:4.1.68.Final]
	at org.apache.bookkeeper.util.ByteBufList.retain(ByteBufList.java:285) ~[classes/:?]
	at org.apache.bookkeeper.client.MockBookKeeperTestCase.lambda$setupBookieClientAddEntry$16(MockBookKeeperTestCase.java:595) ~[test-classes/:?]
	at org.mockito.internal.stubbing.StubbedInvocationMatcher.answer(StubbedInvocationMatcher.java:39) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:96) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:61) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49) ~[mockito-core-3.0.0.jar:?]
	at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptAbstract(MockMethodInterceptor.java:126) ~[mockito-core-3.0.0.jar:?]
	at org.apache.bookkeeper.proto.BookieClient$MockitoMock$249827896.addEntry(Unknown Source) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.sendWriteRequest(PendingAddOp.java:152) ~[classes/:?]
	at org.apache.bookkeeper.client.PendingAddOp.safeRun(PendingAddOp.java:278) ~[classes/:?]
	at org.apache.bookkeeper.common.util.SafeRunnable.run(SafeRunnable.java:36) [classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_292]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_292]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.68.Final.jar:4.1.68.Final]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
